test/parallel/test-runner-run-watch.mjs

```diff
@@ -0,0 +1,187 @@
+// Flags: --expose-internals
+import * as common from '../common/index.mjs';
+import { describe, it, beforeEach } from 'node:test';
+import assert from 'node:assert';
+import { spawn } from 'node:child_process';
+import { writeFileSync, renameSync, unlinkSync, existsSync } from 'node:fs';
+import util from 'internal/util';
+import tmpdir from '../common/tmpdir.js';
+import { join } from 'node:path';
+
+if (common.isIBMi)
+  common.skip('IBMi does not support `fs.watch()`');
+
+// This test updates these files repeatedly,
+// Reading them from disk is unreliable due to race conditions.
+const fixtureContent = {
+  'dependency.js': 'module.exports = {};',
+  'dependency.mjs': 'export const a = 1;',
+  'test.js': `
+const test = require('node:test');
+require('./dependency.js');
+import('./dependency.mjs');
+import('data:text/javascript,');
+test('test has ran');`,
+};
+
+let fixturePaths;
+
+function refresh() {
+  tmpdir.refresh();
+
+  fixturePaths = Object.keys(fixtureContent)
+    .reduce((acc, file) => ({ ...acc, [file]: tmpdir.resolve(file) }), {});
+  Object.entries(fixtureContent)
+    .forEach(([file, content]) => writeFileSync(fixturePaths[file], content));
+}
+
+const runner = join(import.meta.dirname, '..', 'fixtures', 'test-runner-watch.mjs');
+
+async function testWatch({ fileToUpdate, file, action = 'update', cwd = tmpdir.path }) {
+  const ran1 = util.createDeferredPromise();
+  const ran2 = util.createDeferredPromise();
```

Comment:
Can we use `Promise.withResolvers()` and avoid `--expose-internals`?

Reply:
I think it's better to leave them as is to ease of backporting.